src/mongocxx/uri.cpp

```diff
@@ -48,9 +48,12 @@ uri::uri(std::unique_ptr<impl>&& implementation) {
     _impl.reset(implementation.release());
 }
 
-uri::uri(bsoncxx::string::view_or_value uri_string)
-    : _impl(stdx::make_unique<impl>(libmongoc::uri_new(uri_string.terminated().data()))) {
-    if (_impl->uri_t == nullptr) {
+uri::uri(bsoncxx::string::view_or_value uri_string) {
+    bson_error_t error = {0};
+    _impl = stdx::make_unique<impl>(
+        libmongoc::uri_new_with_error(uri_string.terminated().data(), &error));
+
+    if (error.domain) {
```

Comment:
Since [mongoc_uri_new_with_error](http://mongoc.org/libmongoc/1.16.2/mongoc_uri_new_with_error.html) will set `error` to `NULL` on success, this will fail if the function succeeds. What you can do instead is check the result of `mongoc_uri_new_with_error` instead. According to the docs, that function returns `NULL` if it fails. Here's an example from `index_view.hh:197-204`:
```
        bson_error_t error;

        bool result = libmongoc::collection_write_command_with_opts(
            _coll, command_bson.bson(), opts_bson.bson(), reply.bson_for_init(), &error);

        if (!result) {
            throw_exception<operation_exception>(error);
        }
```